atlasdb-perf/src/test/java/com/palantir/atlasdb/performance/PerformanceResultsTest.java

```diff
@@ -54,6 +77,47 @@ public void canGenerateBenchmarkNameForTestWithKeyValueService() {
         assertThat(PerformanceResults.getBenchmarkName(params)).isEqualTo(FORMATTED_BENCHMARK_NAME_CASSANDRA);
     }
 
+    @Test
+    public void doesNotDownsampleSmallSample() {
+        MultisetStatistics stats = new MultisetStatistics();
+        for (double number : SMALL_SAMPLE) {
+            stats.addValue(number, 1);
+        }
+        Mockito.when(mockResult.getStatistics()).thenReturn(stats);
+        assertThat(PerformanceResults.getData(mockRunResult)).containsExactlyElementsOf(SMALL_SAMPLE);
+    }
+
+    @Test
+    public void downSamplesCorrectlyWhenUniform() {
+        MultisetStatistics stats = new MultisetStatistics();
+        for (int i = 0; i < 10; ++i) {
+            for (double number : SMALL_SAMPLE) {
+                stats.addValue(number, 2);
+            }
+        }
+        Mockito.when(mockResult.getStatistics()).thenReturn(stats);
+        assertThat(PerformanceResults.getData(mockRunResult)).containsExactlyElementsOf(SMALL_SAMPLE);
+    }
+
+    @Test
+    public void downSampledDistributionIsRepresentativeForReasonableData() {
+        MultisetStatistics stats = new MultisetStatistics();
+        for (double number : LARGE_SAMPLE) {
+            int elements = (int) (10 * PerformanceResults.DOWNSAMPLE_MAXIMUM_SIZE
+                    / (Math.abs(PerformanceResults.DOWNSAMPLE_MAXIMUM_SIZE - number) + 1));
+            stats.addValue(number,  elements);
+        }
+        Mockito.when(mockResult.getStatistics()).thenReturn(stats);
+        List<Double> downSampledData = PerformanceResults.getData(mockRunResult);
+
+        MultisetStatistics downSampledStats = new MultisetStatistics();
+        for (double number : downSampledData) {
+            downSampledStats.addValue(number, 1);
+        }
+
+        assertThat(stats.compareTo(downSampledStats)).isEqualTo(0);
```

Comment:
Fair enough! I also added a homoscedasticTTest to test variance for good measure. My stats are a bit rusty, but I think the parameters I put in are fine. Wouldn't mind a confirmation though :)